source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -770,6 +770,51 @@ ShaderNodeImplPtr GlslShaderGenerator::createCompoundImplementation(NodeGraphPtr
 }
 
 
+ValuePtr GlslShaderGenerator::remapEnumerationValue(const ValueElementPtr& elem, const InterfaceElement& element, string& elemType)
+{
+    const string& valueElementName = elem->getName();
+    const string& valueString = elem->getValueString();
+    if (valueString.empty())
+    {
+        return nullptr;
+    }
```

Comment:
Not sure about the logic here. If the input has no value set null is returned which will maintain the type of the input created after this call. But if the input has a value the type will be converted to integer. Should the type not be the same regardless of the value. Maybe instead let strings always the converted to integers for GLSL, and just let this method convert the value.

Reply:
This would be the case where the nodedef does not specify a default value which I've only seem occur for framerange currently. The problem with always turning everything into an int is that it's incorrect for something like a swizzle. I can detect filenames but I can't detect a swizzle currently. If I could then I could change the type to always be int but the value is still unknown so guess it could be 0 or any value.

Reply:
Maybe I need to add in a "does this node handle the conversion by itself" query. Then each node could test the element to return true or false.